math/n_choose_r.cpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file
+ * @brief [Combinations](https://en.wikipedia.org/wiki/Combination) n choose r function implementation
+ * @details
+ * A very basic and efficient method of calculating
+ * choosing r from n different choices. For more information visit 
+ * [Combination](https://en.wikipedia.org/wiki/Combination) on Wikipedia.
```

Comment:
No need to add it here once it is on the brief description.
```suggestion
```

Reply:
Okay, I'll change it now